s4e-web/src/app/test.utils.spec.ts

```diff
@@ -0,0 +1,11 @@
+import {Observable} from 'rxjs';
+import {take, toArray} from 'rxjs/operators';
+
+export function toTestPromise<T>(observable: Observable<T>, takeLast: number = 1): Promise<T|T[]> {
+  observable = observable.pipe(take(takeLast))
+
+  if (takeLast === 1) {
+    return observable.toPromise();
+  }
+  return observable.pipe(toArray()).toPromise();
```

Comment:
This can be simplified 

```
 if (takeLast === 1) {
    return observable.toPromise();
  }
  return observable.pipe(toArray()).toPromise();
```

to

```
  return (takeLast === 1 ? observable : observable.pipe(toArray()) ).toPromise();
```